config/src/main/java/org/springframework/security/config/annotation/authentication/configuration/AuthenticationConfiguration.java

```diff
@@ -103,12 +103,10 @@ public static InitializeAuthenticationProviderBeanManagerConfigurer initializeAu
 		return new InitializeAuthenticationProviderBeanManagerConfigurer(context);
 	}
 
-	public AuthenticationManager getAuthenticationManager() throws Exception {
+	public AuthenticationManager getAuthenticationManager(AuthenticationManagerBuilder authBuilder) throws Exception {
```

Comment:
This is a major API change so it could only be allowed in Spring Security 6.0. Is this issue scheduled for 6.0? For more info around versioning see [Semantic Versioning](https://semver.org/spec/v2.0.0.html)

Reply:
I struggle with this a little bit because we want to guarantee that the `AuthenticationManagerBuilder` is actually a `Bean`. Perhaps we could change to do this:

```java
AuthenticationManagerBuilder authBuilder = this.applicationContext.getBean(AuthenticationManagerBuilder.class);
```